utility/src/main/java/org/oppia/android/util/parser/image/UrlImageParser.kt

```diff
@@ -151,19 +151,21 @@ class UrlImageParser private constructor(
 
     override fun onResourceReady(resource: T, transition: Transition<in T>?) {
       val drawable = retrieveDrawable(resource)
+      // This statement is logged correctly
       htmlContentTextView.post {
-        htmlContentTextView.width { viewWidth ->
```

Comment:
This was originally needed to fix an issue where the image may load faster than the TextView being laid out (which isn't completely surprising--post has inherent issues which is why we're considering outright banning it per #4689). See #507. You'll need to reevaluate to make sure that issue can't happen anymore even after this change.

Reply:
I have checked by removing .width {} and it is again introducing the similar issue #507. 


https://user-images.githubusercontent.com/43074241/199662466-9ff924f2-53e2-418e-94bb-94c64855eec2.mp4

Reply:
> @BenHenning instead of using our .width{..} function I have created a new function afterLayout according to this [stackoverflow answer](https://stackoverflow.com/a/57495622/12314934). With this solution both problems are solved
> 
> 1. Loading images in selectionInteractionView
> 2. Issue [Rich text image centre fix #507](https://github.com/oppia/oppia-android/pull/507)
> 
>  fraction-frequently.mp4

Can you provide context on *why* this fixes both issues? I think that's important here (it feels a bit like we just copied the solution from SO, but we need to understand why the approach actually works).

Reply:
Your explanation more or less just repeats _what_ the code is doing @vrajdesai78 but what I'm actually asking for is the _why_. Specifically, why does the width == 0 check fail when using content descriptions? And, why does isLaidOut work to address both that issue & the original issue fixed by this global layout codef?

Please be clear on the 'why'--it's really important as it's the core purpose of this code. Please also capture this in an in-line comment in the code so that future developers can better contextualize the method if future changes need to be made.

Reply:
I have kept the original code of .width {..} method. I have just added requestLayout() before calling onGlobalLayoutListener. Without using requestLayout() onGlobalLayoutListener is not getting called when we have images inside an recyclerview with content description. The reason why it happened is because onGlobalLayout is invoked only when the global layout state changes and it might be possible that the view is created before the request to load images processed (may be the http request). [Stackoverflow](https://stackoverflow.com/a/35444014/12314934) answer to support my observation. Here requestLayout() will help to call onGlobalLayoutListener which solves the problem of loading images with content description in a recyclerview.

Reply:
Thanks @vrajdesai78! That's really helpful--can you please also update the in-line comment to include this context? (I also left one new comment asking the same).

Reply:
Updated in inline comment, thanks.